db/re/item_db_equip.yml

```
@@ -23265,6 +23252,40 @@ Body:
     Script: |
       bonus bInt,2;
       bonus bMatk,(readparam(bInt) >= 120 ? 30 : 5);
+  - Id: 2192
+    AegisName: Rose_Of_Eden
+    Name: Rose of Eden
+    Type: Armor
+    View: 3
+    Weight: 300
+    Defense: 70
+    Slots: 1
+    Locations:
+      Left_Hand: true
+    ArmorLevel: 1
+    EquipLevelMin: 70
+    Refineable: true
+    Script: |
+      bonus bHealPower,10;
+  - Id: 2193
+    AegisName: Fallen_Angel_Shield
+    Name: Fallen Angel Shield
+    Type: Armor
+    View: 4
+    Weight: 500
+    Defense: 100
+    Slots: 1
+    Locations:
+      Left_Hand: true
+    ArmorLevel: 1
+    EquipLevelMin: 60
+    Refineable: true
+    Script: |
+      .@r = getrefine();
+      if (.@r >=7)
+         autobonus "{ bonus bDelayrate,-10-(5*(.@r-6)); }",50,5000,BF_MAGIC;
+      else
+         autobonus "{ bonus bDelayrate,-10; }",50,5000,BF_MAGIC;
```

"When hit by" it should be autobonus2
The `.@r` inside your autobonus is not defined, the bonus inside the string is running independently of the item script
```suggestion
      if (.@r >=7)
         autobonus2 "{ .@r = getrefine(); bonus bDelayrate,-10-(5*(.@r-6)); }",50,5000,BF_MAGIC;
      else
         autobonus2 "{ bonus bDelayrate,-10; }",50,5000,BF_MAGIC;
```

---

Oh my apologies and thanks for the knowledge.